### PR TITLE
chore(deps): adds external context dep

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "eslint-plugin-import": "^2.18.2",
     "eslint-plugin-node": "^11.0.0",
     "eslint-plugin-prettier": "^3.1.1",
+    "isomorphic-ws": "^4.0.1",
```
Comment (andrewxhill) [inline package.json:90]: missing. not sure how this passed without it?

File: src/users.spec.ts [hunk 2]
```diff
@@ -223,8 +219,7 @@ describe('Users...', () => {
         const db = new Client(ctx)
         ctx = ctx.withThreadName('my-buckets')
         const id = ThreadID.fromRandom()
-        db.config = ctx
-        await db.newDB(id.toBytes())
+        await db.newDB(id, ctx)
         ctx = ctx.withThread(id)
```
Comment (carsonfarmer) [inline src/users.spec.ts:223]: This line can be removed now

Comment (carsonfarmer) [reply src/users.spec.ts:223]:
```suggestion
```

File: src/users.spec.ts [hunk 3]
```diff
@@ -194,10 +192,8 @@ describe('Users...', () => {
       // Got one
       ctx = ctx.withThreadName('foo')
       const id = ThreadID.fromRandom()
-      // Update existing db config directly as it doesn't yet directly support Context API on method calls
-      db.config = ctx
-      // @todo: In the near future, this should be `await db.newDB(id, ctx)`
-      await db.newDB(id.toBytes())
+      
+      await db.newDB(id, ctx)
```
Comment (carsonfarmer) [inline src/users.spec.ts:196]:
```suggestion
      await db.newDB(id, ctx.withThreadName('foo'))
```

That changed, coupled with the deletion above feels cleaner to me, and is now possible :)

File: src/users.spec.ts [hunk 4]
```diff
@@ -194,10 +192,8 @@ describe('Users...', () => {
       // Got one
       ctx = ctx.withThreadName('foo')
```
Comment (carsonfarmer) [inline src/users.spec.ts:193]:
```suggestion
```

This plus the change below feels cleaner now...

Comment (carsonfarmer) [reply src/users.spec.ts:193]: Could do the same thing above as well!

File: src/users.spec.ts [hunk 5]
```diff
@@ -109,10 +109,8 @@ describe('Users...', () => {
       // All good
       ctx = ctx.withThreadName('foo')
```
Comment (carsonfarmer) [inline src/users.spec.ts:110]:
```suggestion
```

See comment below

File: src/users.spec.ts [hunk 6]
```diff
@@ -109,10 +109,8 @@ describe('Users...', () => {
...
       const id = ThreadID.fromRandom()
-      // Update existing db config directly as it doesn't yet directly support Context API on method calls
-      db.config = ctx
-      // @todo: In the near future, this should be `await db.newDB(id, ctx)`
-      await db.newDB(id.toBytes())
+      
+      await db.newDB(id, ctx)
```
Comment (carsonfarmer) [inline src/users.spec.ts:113]:
```suggestion
      await db.newDB(id, ctx.withThreadName('foo'))
```

See comment below

File: src/users.spec.ts [hunk 7]
```diff
@@ -77,7 +77,7 @@ describe('Users...', () => {
       ctx = ctx.withThreadName('foo')
```
Comment (carsonfarmer) [inline src/users.spec.ts:77]:
```suggestion
```

See comment below

File: src/users.spec.ts [hunk 8]
```diff
@@ -77,7 +77,7 @@ describe('Users...', () => {
...
       const id = ThreadID.fromRandom()
       const db = new Client(ctx)
-      await db.newDB(id.toBytes())
+      await db.newDB(id)
```
Comment (carsonfarmer) [inline src/users.spec.ts:80]:
```suggestion
      await db.newDB(id, ctx.withThreadName('foo'))
```

File: package.json (hunk 1 above)
Comment (andrewxhill) [reply package.json:90]: hmm... didn't fix it

File: src/users.spec.ts [hunk 9]
```diff
@@ -223,9 +216,7 @@ describe('Users...', () => {
         const db = new Client(ctx)
         ctx = ctx.withThreadName('my-buckets')
```
Comment (carsonfarmer) [inline src/users.spec.ts:217]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (andrewxhill) [reply package.json:90]: installed `ws` as well and it seems to work again

File: src/users.spec.ts [hunk 10]
```diff
@@ -273,7 +264,7 @@ describe('Users...', () => {
...
```
Comment (carsonfarmer) [inline src/users.spec.ts:264]:
```suggestion
```

File: src/users.spec.ts [hunk 11]
```diff
@@ -223,9 +216,7 @@ describe('Users...', () => {
         const db = new Client(ctx)
         ctx = ctx.withThreadName('my-buckets')
         const id = ThreadID.fromRandom()
-        db.config = ctx
-        await db.newDB(id.toBytes())
-        ctx = ctx.withThread(id)
+        await db.newDB(id, ctx)
```
Comment (carsonfarmer) [inline src/users.spec.ts:219]:
```suggestion
        await db.newDB(id, ctx.withThreadName('my-buckets'))
```

File: src/users.spec.ts [hunk 12]
```diff
@@ -273,7 +264,7 @@ describe('Users...', () => {
         ctx = ctx.withThreadName('my-buckets')
         const id = ThreadID.fromRandom()
         const db = new Client(ctx)
-        await db.newDB(id.toBytes())
+        await db.newDB(id)
```
Comment (carsonfarmer) [inline src/users.spec.ts:267]:
```suggestion
        await db.newDB(id, ctx.withThreadName('my-buckets'))
```

**Pull request closed without merging**